### DISCUSSION TITLE
Custom Experimental inputs and output variables.

### DISCUSSION THREAD

**Discussion by balaji3112** [pybamm-team/liionpack#220] [Q&A]
Is it possible to give our own experimental input current data through an excel or CSV file to run the simulation? Also, Is it possible to save/call the simulation output variables separately?

**Comment by TomTranter** [comment]
Example using a drive cycle for current data here https://github.com/pybamm-team/liionpack/blob/develop/examples/drive_cycle.py
Outputs get lumped into 1 dictionary but you could write your own function to unpack and write to csv easily enough

**Reply by balaji3112** [reply]
Thank you.

**Reply by balaji3112** [reply]
I tried using a custom drive_cycle file and it's working fine, but I still want to know what I should give as operating_conditions and drive_cycles in the Experiment class.

drive_cycle = pd.read_csv(
    "pybamm/input/drive_cycles/pulse_discharge.csv", comment="#", header=None
).to_numpy()

experiment = pybamm.Experiment(
    operating_conditions=["Run US06 (A)"],
    period="1 s",
    drive_cycles={"US06": drive_cycle})
